saleor/checkout/calculations.py

```diff
@@ -347,6 +347,25 @@ def update_prior_unit_price_for_lines(lines: Iterable["CheckoutLineInfo"]):
             line_info.line.prior_unit_price_amount = line_info.prior_unit_price_amount
 
 
+def _is_checkout_modified(
+    checkout: "Checkout",
+    database_connection_name: str = settings.DATABASE_CONNECTION_DEFAULT_NAME,
+) -> bool:
+    """Check if the checkout has been modified during recalculation."""
+    checkout_last_change = checkout.last_change
+    try:
+        refreshed_checkout_last_change = (
+            Checkout.objects.using(database_connection_name)
+            .only("last_change")
+            .get(token=checkout.token)
+            .last_change
```

Comment:
As you built this system based on assumption that `last_change` always stores the correct value, the maybe worth to extends the mutation tests to confirm that no matter what, we always set the last_change? 🤔

Reply:
Additional question. Why `last_change` instead of `price_expiration`? I mean `price_expiration` says that "we will re-calculate prices", where `last_change` means "something has been changed". Where `something` can also be a thing not related to the prices 🤔  
Like:
- (Worker1) calling checkoutLinesUpdate - price are expired
- (Worker2) immediately calling checkoutCustomerNoteUpdate - (like system adds some customization to the line that you're going to add) - mutation updates `last_change`
- (Worker1) - `last_change` has beed modified, we skip price save.

Reply:
The goal is not to create incosistent states. The criteria we chose is "was I the last one to modify the object" for which `last_change` is the correct field.

Reply:
If we skip saving, we also won't update the `price_expiration`, so subsequent queries and updates should attempt to recalculate, no?